apps/dashboard/src/components/workflow-editor/url-input.tsx

```diff
@@ -0,0 +1,102 @@
+import { liquid } from '@codemirror/lang-liquid';
+import { EditorView } from '@uiw/react-codemirror';
+import { useFormContext } from 'react-hook-form';
+
+import { Input, InputField, InputFieldProps, InputProps } from '@/components/primitives/input';
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@/components/primitives/select';
+import { Editor } from '@/components/primitives/editor';
+import { FormControl, FormField, FormItem, FormMessagePure } from '@/components/primitives/form/form';
+import { cn } from '@/utils/ui';
+
+type URLInputProps = Omit<InputProps, 'value' | 'onChange' | 'size'> & {
+  options: string[];
+  asEditor?: boolean;
+  urlKey: string;
+  targetKey: string;
+  withHint?: boolean;
+} & Pick<InputFieldProps, 'size'>;
+
+export const URLInput = ({
```

Comment:
I've moved this component from the primitives as it's based on the form fields, and I couldn't make it work without the form fields.

Reply:
Makes sense. Should we create an `elements` directory for components like this? If it makes sense outside of the workflow editor of course.

Reply:
Let's see if there will be a need to reuse this one.